FWCore/MessageLogger/src/JobReport.cc

```diff
@@ -302,18 +302,40 @@ namespace edm {
     }
   }
 
+  static char const* const kJobReportEndElement = "</FrameworkJobReport>\n";
+  static constexpr int kEndElementSize = 22;
+  static constexpr int kMinSizeOfComment = 8;
+
   JobReport::~JobReport() {
     impl_->flushFiles();
     if (impl_->ost_) {
-      *(impl_->ost_) << "</FrameworkJobReport>\n" << std::flush;
+      //are we actually at the end of the file?
+      auto pos = impl_->ost_->tellp();
+      impl_->ost_->seekp(0, std::ios_base::end);
+      auto endpos = impl_->ost_->tellp();
+      impl_->ost_->seekp(pos);
+      if ((endpos - pos) > kEndElementSize) {
+        //need to add some padding so use a comment element
+        auto padding = (endpos - pos) - (kEndElementSize + kMinSizeOfComment);
+        *(impl_->ost_) << "<!--";
```

Comment:
Could you add a comment here explaining briefly why an XML comment is used instead of white space? (to remind us the next time we need to look at this code)